From 5ad66f8f01a41a54bcdee90053a7e0f444bca17c Mon Sep 17 00:00:00 2001
From: orangepi-xunlong <258384131@qq.com>
Date: Thu, 6 Jul 2023 11:32:58 +0800
Subject: [PATCH 52/57] mtd: spi-nor: Support zb25vq16as 2mb spiflash

---
 drivers/mtd/spi-nor/xmc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mtd/spi-nor/xmc.c b/drivers/mtd/spi-nor/xmc.c
index 30da07542b69..29b8ad9922fd 100644
--- a/drivers/mtd/spi-nor/xmc.c
+++ b/drivers/mtd/spi-nor/xmc.c
@@ -16,6 +16,8 @@ static const struct flash_info xmc_nor_parts[] = {
 	{ "XM25QH128A", INFO(0x207018, 0, 64 * 1024, 256)
 		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
 			      SPI_NOR_QUAD_READ) },
+	{ "ZB25VQ16AS", INFO(0x5e6015, 0, 64 * 1024,  32)
+		NO_SFDP_FLAGS(SECT_4K) },
 	{ "ZB25VQ128A", INFO(0x5e4018, 0, 64 * 1024, 256)
 		NO_SFDP_FLAGS(SECT_4K | SPI_NOR_DUAL_READ |
 			      SPI_NOR_QUAD_READ) },
-- 
2.39.5

